homeassistant/helpers/entity_platform.py

```diff
@@ -147,6 +148,10 @@ def __init__(
             self.platform_name, []
         ).append(self)
 
+        self.domain_entities: dict[str, Entity] = hass.data.setdefault(
+            DATA_DOMAIN_ENTITIES, {}
+        ).setdefault(domain, {})
```

Comment:
This dict is the index by domain which is the same dict for all platforms with the same domain. The entity component uses so every `switch.` will end up in the `switch` bucket so we don't have to walk all the platforms every time an entity service call or the update service gets called